src/fetcher.ts

```diff
@@ -81,35 +80,42 @@ export class BaseFetcher {
   ): Promise<T> {
     const controller = new AbortController()
     const id = setTimeout(() => controller.abort(), this.timeout)
-    const response: Response = await fetch(url, {
-      ...options,
-      signal: controller.signal
-    })
 
-    clearTimeout(id)
-    if (!response?.ok) {
-      throw new ApiError(
-        `Request failed with status ${response?.status} and message: ${response?.statusText}`
-      )
-    }
+    try {
+      const response: Response = await fetch(url, {
+        ...options,
+        signal: controller.signal
+      })
+      clearTimeout(id)
+
+      if (!response.ok) {
```

Comment:
```suggestion
      if (!response?.ok) {
```

We can add an optional chaining just in case response will be undefined (i can't imagine this escenario but, just in case)